packages/core/README.md

```diff
@@ -78,117 +62,378 @@ app.all("/mcp", async (c) => {
 });
 ```
 
-## Creating an MCP Server
+> [!TIP]
+>
+> The Model Context Protocol (MCP) is an open standard that enables secure connections between host applications and external data sources and tools, allowing AI assistants to reason over information and execute functions with user permission.
 
-Basic constructor usage:
+## Features
 
-```typescript
-import { McpServer } from "mcp-lite";
+- No runtime dependencies and a single TypeScript entrypoint.
+- Type-safe tool definitions with Standard Schema (Zod, Valibot, Effect, ArkType).
+- Structured outputs with runtime validation and schema exposure via `tools/list`.
+- HTTP + SSE transport built on the Fetch API (no stdio wrapper required).
+- Adapter interfaces for sessions, client requests, and persistence when you outgrow stateless mode.
+- Middleware hooks and server composition via `.group()` for modular setups.
 
-const server = new McpServer({
-  name: "my-server",
-  version: "1.0.0",
-});
+## Installation
+
+```bash
+npm install mcp-lite
+# or
+bun add mcp-lite
+# or
+pnpm add mcp-lite
 ```
 
-### Using a Schema Adapter
+## Type Safety
+
+### Automatic Type Inference
 
-Schema adapters are needed when using Standard Schema validators (like Zod or Valibot) to convert them to JSON Schema format that MCP clients can understand.
+Standard Schema validators provide automatic type inference:
 
-#### With Zod schema adapter:
 ```typescript
 import { z } from "zod";
 
-const server = new McpServer({
-  name: "my-server",
-  version: "1.0.0",
-  schemaAdapter: (schema) => z.toJSONSchema(schema as z.ZodType),
+const SearchSchema = z.object({
+  query: z.string(),
+  limit: z.number().optional(),
+  filters: z.array(z.string()).optional()
 });
+
+server.tool("search", {
+  inputSchema: SearchSchema,
+  handler: (args) => {
+    // args is typed as { query: string, limit?: number, filters?: string[] }
+    args.query.toLowerCase()
+    args.limit ?? 10
+    args.filters?.map(f => f.trim())
+
+    return { content: [{ type: "text", text: "..." }] }
+  }
+})
 ```
 
-#### Without schema adapter (JSON Schema only):
+### Structured Outputs
+
+Tools can return both human-readable content and machine-readable structured data. Use `outputSchema` to define the shape of `structuredContent`:
```

Comment:
would be good to link to the spec docs on `structuredContent` because (as dsp mentioned) a lot of people are unawares of this part of the june spec, and just seeing `structuredContent` show up in the response might not click (i.e., might seem library specific instead of spec-compliant)

https://modelcontextprotocol.io/specification/2025-06-18/server/tools#structured-content